File: todolist/java/TodoList.java
import java.util.ArrayList;
import java.util.Scanner;

public class TodoList {

    static ArrayList<String> todoListTitles = new ArrayList<>();
     static ArrayList<String> todoListDescriptions = new ArrayList<>();
    static ArrayList<Integer> todoListUniqueId = new ArrayList<>();
     static ArrayList<String> todoListCompletionStatus = new ArrayList<>();

    public static void main(String[] args) {

        Scanner inputCollector = new Scanner(System.in);
        int count = 0;

        while (true) {
            System.out.println("");
            System.out.println("Welcome to the TodoList!");
            System.out.println("1. Add Task");
            System.out.println("2. View Task");
            System.out.println("3. Complete Task");
            System.out.println("4. Delete Task");
            System.out.println("");
            System.out.print("Enter operation: ");
            
            int choice = inputCollector.nextInt();
            inputCollector.nextLine();

            if (choice == 1) {
                System.out.println("");
                while (true) {
                    System.out.print("Enter the title of the task: ");
                    String titleTask = inputCollector.nextLine();

                    System.out.print("Enter the description: ");
                    String descriptionTask = inputCollector.nextLine();

                    count++;
                    int idNumber = count;
                    
                    System.out.println("");
                    System.out.println(addTask(titleTask, descriptionTask, idNumber));
                    System.out.println("");

                    System.out.print("Do you want to enter another task (yes/no): ");
                    String answer = inputCollector.nextLine();
                    if (answer.equalsIgnoreCase("no")) {
                        break;
                    }
                    System.out.println("");
                }

            } else if (choice == 2) {
                System.out.println("");
                viewTask();

            } else if (choice == 3) {
                System.out.println("");
                String completeTag = "complete";

                System.out.print("Enter the unique id to mark completed task: ");
                int uniqueId = inputCollector.nextInt();

                System.out.println("");
                System.out.println(completeTask(uniqueId, completeTag));

            } else if (choice == 4) {
                System.out.println("");
                System.out.print("Enter the unique id to delete task: ");
                int uniqueId = inputCollector.nextInt();

                System.out.println("");
                System.out.println(removeTask(uniqueId));
            }
        }
    }

    public static String addTask(String titleTask, String descriptionTask, int idNumber) {
        todoListTitles.add(titleTask);
        todoListDescriptions.add(descriptionTask);
        todoListUniqueId.add(idNumber);
        todoListCompletionStatus.add("incomplete task");

        return "Task added successfully";
    }

    public static void viewTask() {
        if (todoListTitles.isEmpty()) {
            System.out.println("No tasks to display.");
            System.out.println("");
            return;
        }

        for (int index = 0; index < todoListTitles.size(); index++) {
            System.out.println("The unique id is: " + todoListUniqueId.get(index));
            System.out.println("The title is: " + todoListTitles.get(index));
            System.out.println("The description is: " + todoListDescriptions.get(index));
            System.out.println("The completion status: " + todoListCompletionStatus.get(index));
            System.out.println("");
        }
    }

    public static String completeTask(int idNumber, String completeTag) {
        int index = todoListUniqueId.indexOf(idNumber);

        if (index != -1) {
            todoListCompletionStatus.set(index, completeTag);
            return "Marked as complete!";
        }

        return "Not marked as complete";
    }

    public static String removeTask(int idNumber) {
        int index = todoListUniqueId.indexOf(idNumber);

        if (index != -1) {
            todoListTitles.remove(index);
            todoListUniqueId.remove(index);
            todoListDescriptions.remove(index);
            todoListCompletionStatus.remove(index);
            
            return "Deleted successfully";
        }

        return "Not removed";
    }
}

File: todolist/python/todolist.py

todoListTitles = []

todoListDescriptions = []

todoListUniqueId = []

todoListCompletionStatus = []

def addTask(titleTask, descriptionTask, idNumber):
    todoListTitles.append(titleTask)

    todoListDescriptions.append(descriptionTask)
    todoListUniqueId.append(idNumber)

    todoListCompletionStatus.append("incomplete task")

    return "Task added successfully"

def viewTask():
    if len(todoListTitles) == 0:
        print("No tasks to display.")
        print("")
        return

    
    for index in range(len(todoListTitles)):
        print("The unique id is: " + str(todoListUniqueId[index]))

        print("The title is: " + todoListTitles[index])
        print("The description is: " + todoListDescriptions[index])
        print("The completion status: " + todoListCompletionStatus[index])
        print("")

def completeTask(idNumber, completeTag):
   
    if idNumber in todoListUniqueId:
        index = todoListUniqueId.index(idNumber)
        todoListCompletionStatus[index]=completeTag
      
        return "Marked as complete!"

    return "Not marked as complete"

def removeTask(idNumber):
    if idNumber in todoListUniqueId:
        index = todoListUniqueId.index(idNumber)
        
        todoListTitles.pop(index)
        todoListUniqueId.pop(index)
        todoListDescriptions.pop(index)
        todoListCompletionStatus.pop(index)
        
        return "Deleted successfully"

    return "Not removed"

def main():
    count = 0

    while True:
        print("")
        print("Welcome to the TodoList!")
        print("1. Add Task")
        print("2. View Task")
        print("3. Complete Task")
        print("4. Delete Task")
        print("")
        
        choice = input("Enter operation: ")

        if choice == "1":
            print("")
            while True:
                titleTask = input("Enter the title of the task: ")
                descriptionTask = input("Enter the description: ")

                count += 1
                idNumber = count
                
                print("")
                print(addTask(titleTask, descriptionTask, idNumber))
                print("")

                answer = input("Do you want to enter another task (yes/no): ")
                
                if answer.lower() == "no":
                    break
                print("")

        elif choice == "2":
            print("")
            viewTask()

        elif choice == "3":
            print("")
            completeTag = "complete"

           
            uniqueId = int(input("Enter the unique id to mark completed task: "))

            print("")
            print(completeTask(uniqueId, completeTag))

        elif choice == "4":
            print("")
            uniqueId = int(input("Enter the unique id to delete task: "))

            print("")
            print(removeTask(uniqueId))


main();


